packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -73,7 +73,23 @@ export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClo
                 target={item.target}
                 icon={item.icon}
                 active={item.active}
-                onClick={onClick}
+                onClick={(e: React.MouseEvent<HTMLElement>) => {
+                  // We can have both url and onClick and we want to allow user to open the link in new tab/window
+                  const isSpecialKeyPressed = e.ctrlKey || e.metaKey || e.shiftKey;
+                  if (isSpecialKeyPressed && item.url) {
+                    return;
+                  }
+
+                  if (item.onClick) {
+                    e.preventDefault();
+                    item.onClick(e);
+                  }
```

Comment:
this was not needed before, not sure we need this logic now. 

Can you test 
https://github.com/grafana/grafana/pull/31915

Reply:
ok...doing that now

Reply:
Works as expected

Reply:
> this was not needed before, not sure we need this logic now.
Actually, this was the initial logic we had prior to the menu component refactor.

Reply:
Do we remove this logic? @torkelo

Reply:
Can you link to where this was done before ?

Reply:
Here you go: https://github.com/grafana/grafana/blob/a3d765830e7f4783c58e7a9d8713e1289fdbc3ea/packages/grafana-ui/src/components/Menu/Menu.tsx#L93

Reply:
Yes, but that is part of the MenuItem component, why should that be added to ContextMenu? 

But even then I think that code is not needed, tested and open links in new tab works great

Reply:
Alright 👍